File: android/app/src/main/java/com/basicfit/app/data/AuthManager.java
package com.basicfit.app.data;

import android.content.Context;
import android.content.SharedPreferences;

public class AuthManager {
    private static final String PREFS_NAME = "basicfit_auth";
    private static final String KEY_IS_LOGGED_IN = "is_logged_in";
    private static final String KEY_USER_EMAIL = "user_email";
    private static final String KEY_USER_NAME = "user_name";

    private SharedPreferences prefs;

    public AuthManager(Context context) {
        prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    public boolean login(String email, String password) {
        // Simulation d'une connexion (remplacez par votre API)
        if (email != null && password != null && email.contains("@")) {
            prefs.edit()
                .putBoolean(KEY_IS_LOGGED_IN, true)
                .putString(KEY_USER_EMAIL, email)
                .putString(KEY_USER_NAME, "Utilisateur BasicFit")
                .apply();
            return true;
        }
        return false;
    }

    public boolean register(String email, String password) {
        // Simulation d'une inscription (remplacez par votre API)
        if (email != null && password != null && email.contains("@")) {
            prefs.edit()
                .putBoolean(KEY_IS_LOGGED_IN, true)
                .putString(KEY_USER_EMAIL, email)
                .putString(KEY_USER_NAME, "Nouvel utilisateur")
                .apply();
            return true;
        }
        return false;
    }

    public void logout() {
        prefs.edit()
            .putBoolean(KEY_IS_LOGGED_IN, false)
            .remove(KEY_USER_EMAIL)
            .remove(KEY_USER_NAME)
            .apply();
    }

    public boolean isLoggedIn() {
        return prefs.getBoolean(KEY_IS_LOGGED_IN, false);
    }

    public String getUserInfo() {
        if (isLoggedIn()) {
            String email = prefs.getString(KEY_USER_EMAIL, "");
            String name = prefs.getString(KEY_USER_NAME, "");
            return "Bonjour " + name + "\nEmail: " + email;
        }
        return null;
    }
}

File: android/app/src/main/java/com/basicfit/app/data/AuthManager.kt
package com.basicfit.app.data

import android.content.Context
import com.basicfit.app.ApiService
import com.basicfit.app.LoginRequest
import com.basicfit.app.RegisterRequest
import com.basicfit.app.AuthResponse
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class AuthManager(private val context: Context) {

    private val apiService by lazy {
        ApiService.getInstance().apply {
            if (!isApiAvailable()) {
                initialize(context)
            }
        }.getApi()
    }

    /**
     * Connexion utilisateur.
     * @return Result<AuthResponse> – success si l’API répond success=true
     */
    suspend fun login(email: String, password: String): Result<AuthResponse> = withContext(Dispatchers.IO) {
        return@withContext try {
            val response = apiService.login(LoginRequest(email = email, password = password))
            if (response.success && response.token != null) {
                // Sauvegarder le token
                ApiService.getInstance().saveAuthToken(context, response.token)
            }
            Result.success(response)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    /**
     * Inscription utilisateur.
     */
    suspend fun register(
        email: String,
        password: String,
        nom: String,
        prenom: String,
        dateNaissance: String? = null,
        poids: Double? = null,
        taille: Int? = null,
        genre: String? = null,
        objectifSportif: String? = null,
        niveauExperience: String? = null
    ): Result<AuthResponse> = withContext(Dispatchers.IO) {
        val request = RegisterRequest(
            email = email,
            password = password,
            nom = nom,
            prenom = prenom,
            date_naissance = dateNaissance,
            poids = poids,
            taille = taille,
            genre = genre,
            objectif_sportif = objectifSportif,
            niveau_experience = niveauExperience
        )
        return@withContext try {
            val response = apiService.register(request)
            if (response.success && response.token != null) {
                ApiService.getInstance().saveAuthToken(context, response.token)
            }
            Result.success(response)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    fun logout() {
        ApiService.getInstance().clearAuthToken(context)
    }
}
